src/storage/src/source/mysql/replication.rs

```diff
@@ -462,15 +470,168 @@ pub(crate) fn render<G: Scope<Timestamp = GtidPartition>>(
                         upper_cap_set.downgrade(&*new_upper);
 
                         rewinds.retain(|_, (_, req)| {
-                            !PartialOrder::less_than(&req.snapshot_upper, &new_upper)
+                            // We need to retain the rewind requests whose snapshot_upper has
+                            // at least one timestamp such that new_upper is less than that
+                            // timestamp
+                            req.snapshot_upper.iter().any(|ts| new_upper.less_than(ts))
```

Comment:
I'm this new thing is equivalent of `!PartialOrder::less_equal(&req.snapshot_upper, &new_upper)`. I'll bug you in Slack because I am once more, confused :D

Reply:
yes I think it is the equivalent of that, as that also fixes the issue. But I think the current version is more readable and therefore clearer about what we are validating, so would prefer to leave as-is.